sklearn/utils/_indexing.py

```diff
@@ -417,7 +417,15 @@ def _get_column_indices_interchange(X_interchange, key, key_dtype):
     },
     prefer_skip_nested_validation=True,
 )
-def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=None):
+def resample(
+    *arrays,
+    replace=True,
+    n_samples=None,
+    random_state=None,
+    stratify=None,
+    sample_weight=None,
+    use_weights_in_resampling=False,
```

Comment:
We should now remove this last keyword argument and always use the weights for resampling when they are provided.